### ISSUE TITLE
SNS.publish() not compatible with the AWS API.

### ISSUE THREAD

**Issue by ntquyen** [localstack/localstack#597]
According to https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/SNS.html#publish-property. 

```
Subject — (String)
Optional parameter to be used as the "Subject" line when the message is delivered to email endpoints. This field will also be included, if present, in the standard JSON messages delivered to other endpoints.

Constraints: Subjects must be ASCII text that begins with a letter, number, or punctuation mark; must not include line breaks or control characters; and must be less than 100 characters long.
```
so if subject is an empty string, the API will throw error `InvalidParameter`. But with localstack, the API still return OK.

**Comment by whummer** [comment]
Thanks for reporting @ntquyen . Would you be able to create a small pull request to fix this issue? The `sns_listener.py` file would be the starting point: https://github.com/localstack/localstack/blob/master/localstack/services/sns/sns_listener.py#L58-L90 Thanks!

**Comment by whummer** [comment]
Thanks for reporting @ntquyen . This should be fixed in #2169 - can you please give it a try with the `latest` version. Please report here if the problem persists. Thanks
